diff --git a/orchagent/pfcwdorch.cpp b/orchagent/pfcwdorch.cpp
index e0283bf..afef676 100644
--- a/orchagent/pfcwdorch.cpp
+++ b/orchagent/pfcwdorch.cpp
@@ -475,7 +475,7 @@ void PfcWdSwOrch<DropHandler, ForwardHandler>::registerInWdDb(const Port& port,
         string key = getFlexCounterTableKey(sai_serialize_object_id(port.m_port_id));
         vector<FieldValueTuple> fieldValues;
         // Only register lossless tc counters in database.
-        string str = counterIdsToStr(c_portStatIds, &sai_serialize_port_stat);
+        string str = counterIdsToStr(c_portStatIds, (string (*)(sai_port_stat_t))sai_serialize_port_stat);
         string filteredStr = filterPfcCounters(str, losslessTc);
         fieldValues.emplace_back(PORT_COUNTER_ID_LIST, filteredStr);
 
@@ -504,7 +504,7 @@ void PfcWdSwOrch<DropHandler, ForwardHandler>::registerInWdDb(const Port& port,
 
         if (!c_queueStatIds.empty())
         {
-            string str = counterIdsToStr(c_queueStatIds, sai_serialize_queue_stat);
+            string str = counterIdsToStr(c_queueStatIds, (string (*)(sai_queue_stat_t))sai_serialize_queue_stat);
             queueFieldValues.emplace_back(QUEUE_COUNTER_ID_LIST, str);
         }
 
